src/pages/AddPersonalBankAccountPage.js

```diff
@@ -54,108 +51,84 @@ const defaultProps = {
     },
 };
 
-class AddPersonalBankAccountPage extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.validate = this.validate.bind(this);
-        this.submit = this.submit.bind(this);
-        this.exitFlow = this.exitFlow.bind(this);
-
-        this.state = {
-            selectedPlaidAccountID: '',
-        };
-    }
-
-    componentWillUnmount() {
-        BankAccounts.clearPersonalBankAccount();
-    }
+function AddPersonalBankAccountPage({personalBankAccount, plaidData}) {
+    const {translate} = useLocalize();
+    const [selectedPlaidAccountId, setSelectedPlaidAccountId] = useState('');
+    const shouldShowSuccess = lodashGet(personalBankAccount, 'shouldShowSuccess', false);
 
     /**
      * @returns {Object}
      */
-    validate() {
-        return {};
-    }
+    const validateBankAccountForm = () => ({});
 
-    submit() {
-        const selectedPlaidBankAccount = _.findWhere(lodashGet(this.props.plaidData, 'bankAccounts', []), {
-            plaidAccountID: this.state.selectedPlaidAccountID,
+    const submitBankAccountForm = useCallback(() => {
+        const selectedPlaidBankAccount = _.findWhere(lodashGet(plaidData, 'bankAccounts', []), {
+            plaidAccountID: selectedPlaidAccountId,
         });
 
         BankAccounts.addPersonalBankAccount(selectedPlaidBankAccount);
-    }
+    }, [plaidData, selectedPlaidAccountId]);
 
-    exitFlow() {
-        const exitReportID = lodashGet(this.props, 'personalBankAccount.exitReportID');
-        if (exitReportID) {
-            Navigation.dismissModal(exitReportID);
+    const exitFlow = useCallback(() => {
+        if (personalBankAccount.exitReportID) {
+            Navigation.dismissModal(personalBankAccount.exitReportID);
         } else {
             Navigation.goBack(ROUTES.SETTINGS_WALLET);
         }
-    }
-
-    render() {
-        const shouldShowSuccess = lodashGet(this.props, 'personalBankAccount.shouldShowSuccess', false);
-
-        return (
-            <ScreenWrapper
-                includeSafeAreaPaddingBottom={shouldShowSuccess}
-                shouldEnablePickerAvoiding={false}
-                shouldShowOfflineIndicator={false}
-                testID={AddPersonalBankAccountPage.displayName}
-            >
-                <HeaderWithBackButton
-                    title={this.props.translate('bankAccount.addBankAccount')}
-                    onBackButtonPress={this.exitFlow}
+    }, [personalBankAccount.exitReportID]);
+
+    useEffect(() => BankAccounts.clearPersonalBankAccount, []);
```

Comment:
Shouldn't `BankAccounts.clearPersonalBankAccount` be executed in the return callback of useEffect, so that it matches the componentWillUnmount? If not, this executes on mount, not on unmount 🤔

Reply:
Yup that's matches to the componentWillUnmount lifecycle event.

Basically useEffect with none dependencies matches componentDidMount
return of callback for this use useEffect matches component will unmount

https://react.dev/reference/react/useEffect#reference

Reply:
Oh, nvm, you are returning a callback, didn't notice the missing () on the function call. Sorry, LGTM